### PR TITLE
Support Image and File field types in FileItemNormalizer

### LINKED ISSUE #41: Property "display" is unknown when normalizing ImageItem fields
With Beta 16 and File Entity at 4654bdfee2e301b48da70ad9afab441ceca5b879

I'm hitting this line in TypedDataManager

```
throw new \InvalidArgumentException("Property $property_name is unknown.");
```

Wrapping these two lines in a check for ImageItem fixes the issue.

```
if (!$field_item instanceof ImageItem) {
  $data['_embedded'][$field_uri][0]['display'] = $field_item->get('display')->getValue();
  $data['_embedded'][$field_uri][0]['description'] = $field_item->get('description')->getValue();
}
```

The issue is that the normalizer is being applied for ImageItem which does not have those fields.

### PR DESCRIPTION
**Pull request by benjy**
Fixes #41 

I'm not sure this is the right fix but it demonstrates the issue.

### CODE REVIEW

**Comment by Berdir** [comment]
Make sense, but lets just support ImageItem properties too then (title, alt, ...)

**Comment by benjy** [comment]
Yeah that makes sense, how should we distinguish between the two field types in constructValue() ?

**Comment by benjy** [comment]
Also, could we parse the fields from the field item? Maybe using schema() or something similar..?

**Comment by Berdir** [comment]
Yes, we should actually be able to just use getValue() or so, and for constructValue(), maybe just merge $data into $value. That shouldn't break anythin, although I don't exactly know how the normalization API works right now.

**Comment by benjy** [comment]
What do you think to the approach in the last commit? I pushed that before you commented. Can update to just merge in $data if you prefer.

**Comment by benjy** [comment]
Hmm, with this latest change it's nearly right but we get "target_id": "7", in the exports for images.

File: src/Normalizer/FileItemNormalizer.php [hunk 1]
```diff
@@ -41,9 +43,29 @@ public function normalize($field_item, $format = NULL, array $context = array())
     $entity = $field_item->getEntity();
     $field_uri = $this->linkManager->getRelationUri($entity->getEntityTypeId(), $entity->bundle(), $field_name);
 
-    // Set file field-specific values.
-    $data['_embedded'][$field_uri][0]['display'] = $field_item->get('display')->getValue();
-    $data['_embedded'][$field_uri][0]['description'] = $field_item->get('description')->getValue();
+    // Add any field-specific data.
+    foreach ($this->getAdditionalProperties($field_item->getPluginId()) as $field_name) {
```
Comment (Berdir) [inline src/Normalizer/FileItemNormalizer.php]: $field_name is not a field name, it's a property name. 

Yes, target_id is special, but I'd just explicitly exclude that. so just $data['_embedded'][$field_uri][0] = $field_item->getValue() and then unset target_id.

**Comment by benjy** [comment]
Will fix this up tomorrow.

**Comment by Berdir** [comment]
Merged, thanks!

**Pull request merged**